src/primitives/transaction.h

```diff
@@ -283,10 +283,9 @@ inline void SerializeTransaction(TxType& tx, Stream& s, Operation ser_action, in
         if (tx.vin.size() == 0 && (nVersion & SERIALIZE_TRANSACTION_WITNESS)) {
             /* We read a dummy or an empty vin. */
             READWRITE(flags);
-            READWRITE(*const_cast<std::vector<CTxIn>*>(&tx.vin));
-            READWRITE(*const_cast<std::vector<CTxOut>*>(&tx.vout));
-            if (flags == 0 && tx.vin.size() != 0) {
-                throw std::ios_base::failure("Extended transaction format unnecessarily used");
+            if (flags != 0) {
```

Comment:
do we need to handle the case that `flags == 0` for instance the serialization of a transaction with no inputs and no outputs maybe we should explicitly make sure tx.vout is set empty?